x-pack/plugins/security_solution/server/lib/detection_engine/routes/signals/finalize_signals_migration_route.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ReindexResponse } from 'elasticsearch';
+
+import { IRouter } from 'src/core/server';
+import { DETECTION_ENGINE_SIGNALS_FINALIZE_MIGRATION_URL } from '../../../../../common/constants';
+import { finalizeSignalsMigrationSchema } from '../../../../../common/detection_engine/schemas/request/finalize_signals_migration_schema';
+import { buildRouteValidation } from '../../../../utils/build_validation/route_validation';
+import { BadRequestError } from '../../errors/bad_request_error';
+import { getIndexCount } from '../../index/get_index_count';
+import { decodeMigrationToken } from '../../migrations/helpers';
+import { applyMigrationCleanupPolicy } from '../../migrations/migration_cleanup';
+import { replaceSignalsIndexAlias } from '../../migrations/replace_signals_index_alias';
+import { buildSiemResponse, transformError } from '../utils';
+
+interface TaskResponse {
+  completed: boolean;
+  response?: ReindexResponse;
+  task: { description?: string };
+}
+
+export const finalizeSignalsMigrationRoute = (router: IRouter) => {
+  router.post(
+    {
+      path: DETECTION_ENGINE_SIGNALS_FINALIZE_MIGRATION_URL,
+      validate: {
+        body: buildRouteValidation(finalizeSignalsMigrationSchema),
+      },
+      options: {
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+      const esClient = context.core.elasticsearch.client.asCurrentUser;
+      const { migration_token: migrationToken } = request.body;
+
+      try {
+        const appClient = context.securitySolution?.getAppClient();
+        if (!appClient) {
+          return siemResponse.error({ statusCode: 404 });
+        }
+
+        const { destinationIndex, sourceIndex, taskId } = decodeMigrationToken(migrationToken);
+        const { body: task } = await esClient.tasks.get<TaskResponse>({ task_id: taskId });
```

Comment:
Is the task document cleaned up anywhere? Looks like it has to be explicitly cleaned up if reindex is done async: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html#docs-reindex-task-api

Reply:
Good catch: 45a3af7. This is going to be the only circumstance in which we auto-delete task documents: in the case of an abandoned/failed migration, the document will remain as auditing information.